lib/src/presentation/generic_error_widget.dart

```diff
@@ -0,0 +1,133 @@
+  final double? height;
+
+  /// [messageTitle] to show in the body, in place of the default one
+  final List<TextSpan>? messageBody;
+
+  /// [messageTitle] to show on the title, in place of the default one
+  final String? messageTitle;
+
+  /// [recoverCallback] the function to be called to recover from the error.
+  /// Default to [null]
+  final VoidCallback? recoverCallback;
+
+  /// [type] the type of error. Defaults to [noData]
+  final GenericNoDataTypes type;
+
+  final String? headerIconSvgUrl;
+
+  final EdgeInsets? padding;
+
+  final Key? actionKey;
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Padding(
+        padding: padding ?? const EdgeInsets.all(18.0),
+        child: Container(
+          padding: const EdgeInsets.symmetric(horizontal: 20),
+          child: SingleChildScrollView(
+            child: Column(
+              mainAxisAlignment: align ?? MainAxisAlignment.center,
+              children: <Widget>[
+                largeVerticalSizedBox,
+                if (type == GenericNoDataTypes.error)
+                  SvgPicture.asset(
+                    headerIconSvgUrl ?? errorPageImageSvgPath,
+                    height: MediaQuery.of(context).size.height / 3,
+                  ),
+                Center(
+                  child: Column(
+                    children: <Widget>[
+                      Text.rich(
+                        TextSpan(
+                          children: <TextSpan>[
+                            TextSpan(
+                              text: messageTitle ?? getErrorMessage(),
+                              style: veryBoldSize20Text(),
+                            ),
+                            const TextSpan(text: '\n\n'),
+                            ...messageBody ??
+                                <TextSpan>[
+                                  TextSpan(
+                                    text: defaultUserFriendlyMessage,
+                                    style: normalSize16Text(
+                                      const Color(0xff707070),
+                                    ),
+                                  )
+                                ]
+                          ],
+                        ),
+                        textAlign: TextAlign.center,
+                      ),
+                      mediumVerticalSizedBox,
+                      SizedBox(
+                        width: double.infinity,
+                        height: 48,
+                        child: MyAfyaHubPrimaryButton(
+                          customRadius: 8.0,
+                          buttonKey: actionKey,
+                          buttonColor: Theme.of(context).primaryColor,
+                          textColor: Colors.white,
+                          onPressed: recoverCallback,
+                          text: actionText,
+                        ),
+                      ),
+                      largeVerticalSizedBox,
```

Comment:
Is this widget necessary, considering this component is supposed to be plugged into other UI elements?